### DISCUSSION TITLE
How to change the tab size?

### DISCUSSION THREAD

**Discussion by zwyyy456** [LazyVim/LazyVim#454] [Q&A]
I have tried to edit the `~/.config/nvim/lua/options.lua` and add `vim.opt.tabstop = 4`，but it doesn't work(I didn't require options manually).  The tabsize is still 2. I have also tried to edit the `~/.local/share/nvim/lazy/LazyVim/lua/lazyvim/config/options.lua` and set `opt.tabstop = 4`, it doesn't work, neither.

Do the settings in `~/.config/nvim/lua/options.lua`  have the highest priority?

I am a novice in neovim. Can anyone tell me how to change the tabsize? I would be very grateful.

**Comment by folke** [comment]
Some filetypes have builtin Neovim ftplugins that override those settings, so that might be the reason.

`:verbose set tabstop?`

You can use a filetype autocmd to override those set by ftplugins

**Reply by zwyyy456** [reply]
I tryied to run `:verbose set tabstop?` when `init.lua` or `test.c` is opened,  it shows: `tabstop = 2, last set from Lua`. So that may be the reason. 

Can you go into more detail about this solution at your convenience?

**Reply by folke** [reply]
The files is wrong. It should be in `~/.config/nvim/lua/config/options.lua` Norice the `config`

**Reply by zwyyy456** [reply]
Sorry, It's just my typo error. I haved found the answer, I should change `~/AppData/Local/nvim/lua/config/options.lua` in win11. Sorry to bother you and thanks for your reply.

**Reply by hyuse202** [reply]
When I run `:verbose set tabstop?`, it shows `tabstop = 4, last set from Lua` but the tabsize's still 2.

**Reply by dpetka2001** [reply]
Have you tried adding a file for the filetype you're interested in `after/ftplugin`, so that it can pick up the options for the filetype from there?

**Reply by hyuse202** [reply]
Sorry where is 'after/ftplugin' ?

**Reply by dpetka2001** [reply]
It doesn't exist by default, you have to create it. That's where Neovim usually expects filetype plugin configurations. The structure should be something like this
```
├── after
│   └── ftplugin
│       └── c.lua
├── init.lua
├── lazy-lock.json
├── LICENSE
├── lua
│   ├── config
│   │   ├── autocmds.lua
│   │   ├── keymaps.lua
│   │   ├── lazy.lua
│   │   └── options.lua
│   └── plugins
│       ├── coding.lua
│       ├── colorscheme.lua
│       ├── comments.lua
│       ├── countdown.lua
│       ├── dap.lua
│       ├── editor.lua
```
Directory `after` should be at the root of your Neovim installation.

**Reply by hyuse202** [reply]
Thanks for help.
I fixed it by `set shiftwidth = 4`
